union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
```

Comment:
Is it really meaningless? I'd say the union of zero filesystems is an empty filesystem, just like the union of zero sets is the empty set (and a filesystem is a set of files). Of course write access should fail. Maybe something with `rwvfs.ReadOnly`?